### PR TITLE
feat(chat): add download link to attachments

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* Fix: https://github.com/nextcloud/talk-desktop/issues/824

### 🖼️ Screenshots / Screencasts

image

### 🏁 Checklist

- [x] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [x] Talk Desktop
  - [ ] Not risky to browser differences / client
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 1]
```diff
@@ -106,13 +106,21 @@
...
-						{{ t('spreed', 'Go to file') }}
-					</NcActionLink>
+					<template v-if="isFileShare">
+						<NcActionSeparator />
+						<NcActionLink :href="linkToFile">
+							<template #icon>
+								<File :size="20" />
+							</template>
+							{{ t('spreed', 'Go to file') }}
+						</NcActionLink>
```
Comment (DorraJaouad) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:116]: This wasn't supposed to be shown for guests, no 🤔 ?

Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:116]: Is wasn't shown, but I don't think it wasn't supposed to. It was a bug.

There was no link in the menu, but the preview itself is a link anyway.

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 2]
```diff
@@ -438,13 +450,22 @@ export default {
...
+			return this.messageFile.link
+		},
+
+		linkToFileDownload() {
+			const uid = getCurrentUser()?.uid
+			if (!uid) {
+				const fileid = this.linkToFile.slice(this.linkToFile.lastIndexOf('/') + 1)
+				return this.messageFile.link + '/download/' + encodeURI(fileid)
 			}
+			return davRemoteURL + '/files/' + encodeURI(uid) + '/' + encodeURI(this.messageFile.path)
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:468]: Maybe we'll move it to utils/fileDownload.ts? Feels like a fit to keep reusable logic there

**Review by Antreesy — APPROVED** [review]
Nice addition, tested from my side

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:116]: Shared files in public rooms are public shares, so should be correct here:
- as user: `https://nextcloud.local/index.php/f/585`
- as guest: `https://nextcloud.local/index.php/s/A4ERJc3xMKmN88o`

**Review by DorraJaouad — APPROVED** [review]
Tested ✅

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 3]
```diff
@@ -438,13 +450,15 @@ export default {
...
-			} else {
-				return ''
-			}
+		messageFile() {
+			const firstFileKey = (Object.keys(this.message.messageParameters).find(key => key.startsWith('file')))
+			return this.message.messageParameters[firstFileKey]
+		},
+
+		linkToFileDownload() {
+			return getCurrentUser()
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:459]: So, it's called twice for exisitng user on each file?
I would expect this check to be called once in davUtils

**Review by Antreesy — APPROVED** [review]
But works anyway

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue (hunk 3 above)
Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:459]: The result is already cached by `getCurrentUser()`:

https://github.com/nextcloud-libraries/nextcloud-auth/blob/67f56fa1e23b2451b4675548b650c81f8e452002/lib/user.ts#L30-L33

**Comment by ShGKme** [comment]
Rebased, squashed

**Comment by szaimen** [comment]
🎉🎉🎉

**Pull request merged**